rfcs/0095-enable-docheck-by-default.md

```diff
@@ -0,0 +1,69 @@
+---
+feature: enable-docheck-by-default
+start-date: 2021-06-24
+author: gytis-ivaskevicius
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Enable `doCheck` by default when using `stdenv.mkDerivation` function.
+
+# Motivation
+[motivation]: #motivation
+
+I believe that an additional quality gate would be beneficial to the derivations build process.
+
+Resolution of this RFC is expected to remove [these comments](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix#L61-L67)
+either by enabling checkPhase by default or rejecting this RFC.
+
+# Detailed design
+[design]: #detailed-design
+
+The basic idea is quite simple
+- New `doCheck` semantic should be implemented.
+- By default `doCheck` option should be enabled as long as `stdenv.hostPlatform == stdenv.buildPlatform`.
+- Non-reproducible test prevention should be implemented.
+- All failing packages should be fixed or updated with `doCheck = false;`
+
+**New doCheck semantics:**
+Instead of boolean it should accept these values:
```

Comment:
This is unnecessarily complicated. `doCheck` should be kept as a Boolean. I don't think it's a problem to have to write e.g.
```
doCheck = false;
doInstallCheck = true;
```

Reply:
Alright, maybe that's a little overengineering. Just thought that 1 is simpler than 2.

Would it be fine if a random string would be considered as `false`? 
I think it would be convenient to have some Nix evaluable comment which would work as a reason why check is disabled

Reply:
> I think it would be convenient to have some Nix evaluable comment which would work as a reason why check is disabled

I think being explicit is superior in this case: `doCheck = lib.something "comment" false;`

`something` &rarr; `lib.disableReason` ?

Reply:
Not sure what `lib.something` is supposed  to return, it's just a value, there is no fancy evaluation since it's not a module